### PR TITLE
feat: add Baidu LLM ERNIE

### PR DESCRIPTION
**Pull request by sssvip**
### PR Checklist

- [*] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [*] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [*] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [*] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [*] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [*] Describes the source of new concepts.
- [*] References existing implementations as appropriate.
- [*] Contains test coverage for new functions.
- [*] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

**Comment by sssvip** [comment]
INFO [runner] linters took 34.4039258s with stages: goanalysis_metalinter: 34.3403054s 
::error file=textsplitter/split_documents.go,line=90,col=18::G602: Potentially accessing slice out of bounds (gosec)
::error file=textsplitter/split_documents.go,line=95,col=18::G602: Potentially accessing slice out of bounds (gosec)
INFO File cache stats: 216 entries of total size 481.6KiB 
INFO Memory: 390 samples, avg is 434.4MB, max is 1147.8MB

File: embeddings/ernie/ernie.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+// Ernie https://cloud.baidu.com/doc/WENXINWORKSHOP/s/alj562vvu
+type Ernie struct {
+	client *ernie.LLM
+}
+
+var _ embeddings.Embedder = &Ernie{}
+
+// todo: use option pass, more: https://cloud.baidu.com/doc/WENXINWORKSHOP/s/alj562vvu#body%E5%8F%82%E6%95%B0
+const batchSize = 16
```
Comment (tmc) [inline embeddings/ernie/ernie.go:18]: can you make this an option func?

Comment (sssvip) [reply embeddings/ernie/ernie.go]: Added

File: embeddings/ernie/ernie.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		if err != nil {
+			return nil, err
+		}
+		v.client = client
+	}
+
+	return v, nil
+}
+
+// EmbedDocuments implements embeddings.Embedder .
```
Comment (tmc) [inline embeddings/ernie/ernie.go:41]: this comment should be improved

**Review by tmc — APPROVED** [review]
LGTM, please do a pass on the comments to improve them and include a package-level doc.go when appropriate.

**Comment by sssvip** [comment]
> LGTM, please do a pass on the comments to improve them and include a package-level doc.go when appropriate.

@tmc Okay, I will submit it soon.

**Comment by chenjisuan** [comment]
@sssvip ernie embedder split text by batchsize，may result in len(emb) != len(texts)，and vector store will fail on [here](https://github.com/tmc/langchaingo/blob/72b07a1c060f09c5eb406020d8405c42b2503547/vectorstores/weaviate/weaviate.go#L102)
```
if len(vectors) != len(docs) {
    return ErrEmbedderWrongNumberVectors
}
```
other embedder will do a combinevector conversion, eg:[link](https://github.com/tmc/langchaingo/blob/72b07a1c060f09c5eb406020d8405c42b2503547/embeddings/huggingface/huggingface.go#L39)
```
func (e *Huggingface) EmbedDocuments(ctx context.Context, texts []string) ([][]float64, error) {
	batchedTexts := embeddings.BatchTexts(
		embeddings.MaybeRemoveNewLines(texts, e.StripNewLines),
		e.BatchSize,
	)
	emb := make([][]float64, 0, len(texts))
	for _, texts := range batchedTexts {
		curTextEmbeddings, err := e.client.CreateEmbedding(ctx, texts, e.Model, e.Task)
		if err != nil {
			return nil, err
		}
		textLengths := make([]int, 0, len(texts))
		for _, text := range texts {
			textLengths = append(textLengths, len(text))
		}
		combined, err := embeddings.CombineVectors(curTextEmbeddings, textLengths)
		if err != nil {
			return nil, err
		}
		emb = append(emb, combined)
	}
	return emb, nil
}
```

英语有点拉胯，希望能表达清楚。

**Comment by sssvip** [comment]
@chenjisuan Thank you. I will fix it as soon as possible.

**Pull request merged**